serf/config.go

```diff
@@ -271,6 +273,16 @@ func (c *Config) Init() {
 // for most of the configurations.
 func DefaultConfig() *Config {
 	hostname, err := os.Hostname()
+	//TODO(schristoff): is it safe to add this here? also should be flag
+	if len(hostname) > 128 {
+		err = fmt.Errorf("Hostname is %v characters. Valid length is between 1 and 128 characters", len(hostname))
+	}
```

Comment:
I think this is ok to validate here - we should use the length constant from above